hera_cal/lstbin.py

```diff
@@ -246,15 +225,11 @@ def lst_bin(data_list, lst_list, flags_list=None, dlst=None, lst_start=None, lst
 
     # return un-averaged data if desired
     if return_no_avg:
-        # return all binned data instead of just bin average
-        data_bin = odict(list(map(lambda k: (k, np.array(list(odict(list(map(lambda k2: (k2, data[k][k2]),
-                                                                             sorted(data[k].keys())))).values()))),
-                                  sorted(data.keys()))))
-        flags_bin = odict(list(map(lambda k: (k, np.array(list(odict(list(map(lambda k2: (k2, flags[k][k2]),
-                                                                              sorted(flags[k].keys())))).values()))),
-                                   sorted(flags.keys()))))
+        # return all binned data instead of just the bin average
+        data_bins = odict([(k1, [data[k1][k2] for k2 in data[k1].keys()]) for k1 in data.keys()])
```

Comment:
Before you cast these to numpy arrays (which I believe also makes a copy of the data). Is it OK that you're no longer copying and that you've got a list of arrays rather than a higher dimensional array?

Reply:
in this case yes b/c this line is really only used for debugging, or if the user wants to get higher order statistics from the data bins themselves